cloudwatch_metrics_collector/src/metrics_collector_parameter_helper.cpp

```diff
@@ -161,16 +163,42 @@ void ReadStorageResolution(
 
 void ReadTopics(
         std::shared_ptr<Aws::Client::ParameterReaderInterface> parameter_reader,
-        std::vector<std::string> & topics) {
+        std::vector<TopicInfo> & topics) {
+
+  std::vector<std::string> topic_names;
+  parameter_reader->ReadParam(ParameterPath(kNodeParamMonitorTopicsListKey), topic_names);
 
-  std::string param_key;
-  if (AWS_ERR_OK != parameter_reader->ReadParam(ParameterPath(kNodeParamMonitorTopicsListKey), param_key)) {
-    parameter_reader->ReadParam(ParameterPath(param_key), topics);
+  std::vector<int64_t> topic_types;
+  parameter_reader->ReadParam(ParameterPath(kNodeParamMonitorTopicTypesListKey), topic_names);
```

Comment:
If I run `ros2 param set ...` twice to set those two parameters, is it guaranteed this function will never be called in the meantime? How is this working knowing that by default setting two independent parameters is not atomic?

Reply:
This two param system seems suboptimal. Can't we have two params? One with a list of ros_monitoring_msgs, one with a list of system_metrics_collector messages?

Reply:
So apparently it's an AWS parameter, not a ROS one - I think this feedback is irrelevant. Keeping that here just as an FYI.

Reply:
> This two param system seems suboptimal. Can't we have two params? One with a list of ros_monitoring_msgs, one with a list of system_metrics_collector messages?

I think this is a good idea, and then I won't need to introduce a `aws_monitored_metric_topic_types` parameter, but instead just introduce another parameter of the same type as `aws_monitored_metric_topics`.

Reply:
> If I run `ros2 param set ...` twice to set those two parameters, is it guaranteed this function will never be called in the meantime? How is this working knowing that by default setting two independent parameters is not atomic?

`parameter_reader->ReadParam()` is just a wrapper around `node->get_parameter()`. I'm not sure what special way you might mean when using `node->get_parameter()` to avoid some atomicity problem.